packages/core/src/validate.ts

```diff
@@ -120,57 +121,75 @@ export function validate(solcOutput: SolcOutput, decodeSrc: SrcDecoder): Validat
   return validation;
 }
 
-export function getContractVersion(validation: Validation, contractName: string): Version {
-  const { version } = validation[contractName];
+export function getContractVersion(validations: ValidationLog, contractName: string, compilationRun: number): Version {
+  const { version } = validations[compilationRun][contractName];
   if (version === undefined) {
     throw new Error(`Contract ${contractName} is abstract`);
   }
   return version;
 }
 
-function getContractName(validation: Validation, version: Version): string {
-  const contractName = Object.keys(validation).find(
-    name => validation[name].version?.withMetadata === version.withMetadata,
-  );
+function getContractNameAndCompilerRun(validations: ValidationLog, version: Version): [string, number] {
+  let compilerRun = 0;
+  let contractName;
+
+  for (const [index, validation] of validations.entries()) {
+    contractName = Object.keys(validation).find(
+      name => validation[name].version?.withMetadata === version.withMetadata,
+    );
+    if (contractName !== undefined) {
+      compilerRun = index;
+      break;
+    }
+  }
 
   if (contractName === undefined) {
     throw new Error('The requested contract was not found. Make sure the source code is available for compilation');
   }
 
-  return contractName;
+  return [contractName, compilerRun];
 }
 
-export function getStorageLayout(validation: Validation, version: Version): StorageLayout {
-  const contractName = getContractName(validation, version);
-  const c = validation[contractName];
+export function getStorageLayout(validations: ValidationLog, version: Version): StorageLayout {
+  const [contractName, compilerRun] = getContractNameAndCompilerRun(validations, version);
+  const c = validations[compilerRun][contractName];
   const layout: StorageLayout = { storage: [], types: {} };
   for (const name of [contractName].concat(c.inherit)) {
-    layout.storage.unshift(...validation[name].layout.storage);
-    Object.assign(layout.types, validation[name].layout.types);
+    layout.storage.unshift(...validations[compilerRun][name].layout.storage);
+    Object.assign(layout.types, validations[compilerRun][name].layout.types);
   }
   return layout;
 }
 
-export function getUnlinkedBytecode(validation: Validation, bytecode: string): string {
-  const linkableContracts = Object.keys(validation).filter(name => validation[name].linkReferences.length > 0);
+export function getUnlinkedBytecode(validations: ValidationLog, bytecode: string): string {
```

Comment:
This function wasn't updated correctly. It's meant to look through all of the available validations. It should look through the entire validation log.

I think the correct solution is to iterate over each `validation` in the log running the same code we had before.

```suggestion
for (const validation of ValidationLog) {
  // the body of the function as it was before this PR
}
```